optax/_src/transform_test.py

```diff
@@ -269,6 +270,83 @@ def test_add_noise_has_correct_variance_scaling(self):
 
       chex.assert_tree_all_close(updates_i, updates_i_rescaled, rtol=1e-4)
 
+  def test_sherman_morrison(self):
+    rng = jax.random.PRNGKey(42)
+    rng, sub_rng = jax.random.split(rng)
+    a = jax.random.normal(sub_rng, (3, 3))
+
+    rng, sub_rng = jax.random.split(rng)
+    u = jax.random.normal(sub_rng, (3,))
+
+    ref = jnp.linalg.inv(a + jnp.outer(u, u))
+    a_inv = jnp.linalg.inv(a)
+    a_inv = transform.sherman_morrison(a_inv, u)
+    np.testing.assert_allclose(a_inv, ref, rtol=1e-5)
+
+  def test_scale_by_online_newton_step(self):
+    eps = 5.
+    step = scale_by_online_newton_step(eps)
+    rng = jax.random.PRNGKey(42)
+
+    rng, sub_rng = jax.random.split(rng)
+    w = jax.random.normal(sub_rng, (3,))
+
+    params = {'weights': w}
+
+    def fun(params, x):
+      return 0.5 * (params['weights'] @ x)**2
+
+    rng, sub_rng = jax.random.split(rng)
+    x = jax.random.normal(sub_rng, (3,))
+
+    grad = jax.grad(fun)(params, x)
+
+    state = step.init(params)
+    updates, state = step.update(grad, state)
+
+    u = (w @ x) * x
+
+    # check that step.update applied the sherman_morrison
+    ref = transform.sherman_morrison(jnp.eye(3) / eps, u) @ u
```

Comment:
I think these three lines of the test can be removed for a unit test. The last part of the test is great: it checks the calculated gradients against the analytic calculation that we promise to the user. Whether or not sherman_morrison gets called is more of an implementation detail (indeed I think it could be a private function) and testing that isn't typical for a unit test.